bot/exts/info/information.py

```diff
@@ -354,6 +359,52 @@ async def user_nomination_counts(self, user: FetchedMember) -> Tuple[str, str]:
 
         return "Nominations", "\n".join(output)
 
+    @staticmethod
+    def embed_field_value(embed: Embed, header: str) -> str:
+        """Get the value of a field from an embed with the appropriate header."""
+        for embed_proxy in embed.fields:
+            if embed_proxy.name == header:
+                return embed_proxy.value
+
+        return ""
+
+    async def user_info_menu(self, ctx: Context, message: Message, user: FetchedMember) -> None:
+        """Creates and manages a reaction-based menu for the user info embed."""
+        def event_check(reaction_: Reaction, user_: Member) -> bool:
+            """Make sure that this reaction is what we want to operate on."""
+            return str(reaction_.emoji) in options and user_.id != ctx.bot.user.id
+
+        options = list()
+        if is_mod_channel(ctx.channel):
+            # Allow an infraction search if the user has infractions.
+            if self.embed_field_value(message.embeds[0], "Infractions") != "No infractions":
+                options.append(INFR_EMOJI)
+
+        if options:
+            # Add all the applicable reactions to the message
+            for reaction in options:
+                log.trace(f"Adding reaction: {repr(reaction)}")
+                await message.add_reaction(reaction)
+
+            while True:
+                try:
+                    reaction, reacter = await ctx.bot.wait_for("reaction_add", timeout=60, check=event_check)
+                    log.trace(f"Got reaction: {reaction}")
+                except asyncio.TimeoutError:
+                    log.debug("Timed out waiting for a reaction")
+                    break  # We're done, no reactions within the timeout period
+
+                await message.remove_reaction(reaction.emoji, reacter)
+
+                if reaction.emoji == INFR_EMOJI:
+                    search = ctx.bot.get_command("infraction search")
+                    if search:
+                        asyncio.create_task(search(ctx, user.id))
+
+            log.debug("Ending user menu and clearing reactions.")
+            with suppress(NotFound):
+                await message.clear_reactions()
```

Comment:
This seems pretty overcomplicated and could be simplified, there are no other features that have been discussed which would benefit from this.

I think this could be simplified to drop the while loop and instead do something along the lines of:
```py
if is_mod_channel():
    try:
        # wait for reaction
    except asyncio.TimeoutError:
        # No reaction added in time, log
    finally:
        # Remove all reactions and perform infraction search
```

Reply:
Well, I don't want to search in the finally clause because I don't want it pop in when the menu times out, but I'll add it in an else clause.